Abhi-AnimeX/app/src/main/java/com/abhijith/animex/ui/screens/animelist/AnimeList.kt

```diff
@@ -1,14 +1,61 @@
+package com.abhijith.animex.ui.screens.animelist
+
+import android.net.Uri
 import androidx.compose.foundation.lazy.LazyColumn
 import androidx.compose.runtime.Composable
-import com.abhijith.animex.ui.screens.animelist.AnimeListItem
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.navigation.NavController
+import com.abhijith.animex.domain.model.AnimeItem
+import com.abhijith.animex.domain.usecases.GetAnimeListUseCase
+import com.abhijith.animex.ui.navigation.Screen
+import com.abhijith.animex.ui.screens.animelist.viewmodel.AnimeListViewModel
+import com.abhijith.animex.ui.screens.animelist.viewmodel.AnimeListViewModelFactory
+import com.abhijith.animex.ui.screens.error.ErrorScreen
+import com.abhijith.animex.ui.screens.loading.LoadingScreen
+import com.google.gson.Gson
 
 @Composable
-fun AnimeList() {
-    val names = List(15) { "$it" }
+fun AnimeList(navController: NavController) {
+
+    // TODO to update this once DI is added to the project
+    val getAnimeListUseCase = GetAnimeListUseCase()
+    val factory = AnimeListViewModelFactory(getAnimeListUseCase)
+    val viewModel: AnimeListViewModel = viewModel(factory = factory)
+
+    val navigationEvent by viewModel.navigationEvent.collectAsState()
+
+    // TODO to move this out of this composable
+    LaunchedEffect(navigationEvent) {
+        navigationEvent?.let { anime ->
+            val animeJson = Uri.encode(Gson().toJson(anime))
```

Comment:
why gson stuff iin UI layer?

Reply:
not sure where do i include this part as we need to deserialize the json string